src/pytest_servers/fixtures.py

```diff
@@ -59,3 +63,18 @@ def tmp_upath(request):
     elif param == "s3":
         return request.getfixturevalue("s3path")
     raise ValueError(f"unknown {param=}")
+
+
+@pytest.fixture
+def make_tmp_upath(request: "FixtureRequest"):
+    """Factory function for temporary directories on different filesystems."""
+
+    def _make_tmp(fs: str = "local"):
+        if fs == "local":
+            return request.getfixturevalue("local_path")
+        elif fs == "s3":
+            return request.getfixturevalue("s3path")
```

Comment:
If I understand correctly, with a default scope of those fixtures, calling `make_tmp_upath` twice will give you the same result, which defeats the purpose of the factory.